frontend/src/components/input.tsx

```diff
@@ -52,26 +58,59 @@ export const Input = ({ sendMessage, waiting, suggestions }: InputProps) => {
     [sendMessage, userInput, waiting],
   );
 
+  const uploadFile = async (file: File) => {
+    setIsUploading(true);
+    setIsUploaded(false);
+
+    try {
+      const formData = new FormData();
+      formData.append('file', file);
+      const response = await fetch(`${process.env.BACKEND_URL}/uploadfile`, {
+        method: 'POST',
+        body: formData,
+        credentials: 'include',
+      });
+      if (!response.ok)
+        throw new Error(`Upload failed with status ${response.status}`);
+
+      const { filename, id } = await response.json();
```

Comment:
Could we move this request and response bit to `server.ts` to match the other api calls?

Reply:
Done :)